test/test_integration_variable.py

```diff
@@ -0,0 +1,225 @@
+        Utils.set_variable(self, 'one/password', expected_value)
+        output = self.invoke_cli(self.cli_auth_params,
+                                 ['variable', 'get', '--id=one/password'], exit_code=0)
+        self.assertIn(expected_value, output)
+
+    @integration_test
+    def test_variable_get_short_variable_returns_variable_value(self):
+        expected_value = uuid.uuid4().hex
+        Utils.set_variable(self, 'one/password', expected_value)
+        output = self.invoke_cli(self.cli_auth_params,
+                                 ['variable', 'get', '-i', 'one/password'], exit_code=0)
+        self.assertIn(expected_value, output)
+
+    @integration_test
+    def test_variable_get_with_special_chars_returns_special_chars(self):
+        self.invoke_cli(self.cli_auth_params,
+                       ['policy', 'replace', '-b', 'root', '-f', self.environment.path_provider.get_policy_path("variable")])
+        Utils.set_variable(self, 'variablespecialchars', '"[]{}#@^&<>~\/''\/?\;\';\'"')
+        output = Utils.get_variable(self, 'variablespecialchars')
+        self.assertEquals(output.strip(), '"[]{}#@^&<>~\/''\/?\;\';\'"')
+
+    @integration_test
+    def test_variable_get_variable_has_spaces_returns_variable_value(self):
+        self.invoke_cli(self.cli_auth_params,
+                       ['policy', 'replace', '-b', 'root', '-f', self.environment.path_provider.get_policy_path("variable")])
+        Utils.assert_set_and_get(self, "some variable with spaces")
+
+    @integration_test
+    def test_unknown_variable_raises_not_found_error(self):
+        output = self.invoke_cli(self.cli_auth_params,
+                                 ['variable', 'get', '-i', 'unknown'], exit_code=1)
+        self.assertRegex(output, "404 Client Error: Not Found for url:")
+
+    @integration_test
+    def test_cli_can_batch_get_multiple_variables(self):
+        policy, variables = Utils.generate_policy_string(self)
+        file_name=os.path.join(tempfile.gettempdir(), os.urandom(24).hex())
+        with open(file_name, 'w+b') as temp_policy_file:
+            temp_policy_file.write(policy.encode('utf-8'))
+            temp_policy_file.flush()
+
+            Utils.load_policy(self, temp_policy_file.name)
+        value_map = {}
+        for variable in variables:
+            value = uuid.uuid4().hex
+            Utils.set_variable(self, variable, value)
+            value_map[variable] = value
+
+        batch_result_string = Utils.get_variable(self, *variables)
+        batch_result = json.loads(batch_result_string)
+
+        for variable_name, variable_value in value_map.items():
+            self.assertEquals(variable_value, batch_result[variable_name])
+
+        os.remove(file_name)
+
+    @integration_test
+    def test_batch_existing_and_nonexistent_variable_raises_error(self):
+        output = self.invoke_cli(self.cli_auth_params,
+                                 ['variable', 'get', '-i', 'one/password,unknown'], exit_code=1)
```

Comment:
should be 'one/password unknown' otherwise it will try to verify it as one parameter with "," char
Example : `"Variable 'one/password,elad' not found in account 'cucumber'`

Reply:
updating

Reply:
> You did a great job a lot of tests were added .
> One of the tests under integration_list (test_list_string_offset_raises_error) is not working well. Need to open bug for it and comment this test.
> 
> Left some comments.(went over integration tests)

Thanks! I opened a [ticket](https://github.com/cyberark/conjur/issues/1997) for this and adding it as a comment above the newly commented out test